superset/assets/javascripts/explorev2/stores/store.js

```diff
@@ -1718,3 +1718,9 @@ export function initialState(vizType = 'table') {
     isStarred: false,
   };
 }
+
+// Control Panel fields that re-render chart without need for 'Query button'
+export const autoQueryFields = [
+  'datasource',
```

Comment:
if the datasource changes, don't we need to make a request to the server to get new data?